refactor(admins): disable full name editing in update flow

synapsetron opened the pull request:
This PR introduces a series of refactorings aimed at improving the update functionality for MinistryAdmin, RegionAdmin, and AreaAdmin entities. It introduces specialized DTOs for update scenarios, allowing partial updates while maintaining strict validation for creation flows. Additionally, it configures Bearer authentication as the default scheme for all controllers in AuthCommon.

♻️ **refactor(ministry-admin)**
Replaced MinistryAdminBaseDto with MinistryAdminBaseUpdateDto for update endpoints.

MinistryAdminBaseUpdateDto overrides required fields to allow partial updates.

Updated CommonMinistryAdminService to ignore unused fields during updates.

Adjusted AutoMapper configuration and added mapping rules.

Updated and verified related tests.

♻️ **refactor(area-admin)**
Introduced AreaAdminBaseUpdateDto for more flexible updates.

Controller and service layer now use BaseUpdateUserDto as input.

Added mapping from BaseUpdateUserDto to AreaAdminBaseUpdateDto.

Refactored IAreaAdminService to reflect DTO changes.

Updated relevant unit tests and added minor documentation/formatting improvements.

♻️**refactor(region-admin)**
Created RegionAdminBaseUpdateDto to override required fields from base class and support partial updates.

Updated RegionAdminController to use BaseUpdateUserDto as update model.

Refactored IRegionAdminService, AreaAdminService and update-related methods.

Added and updated AutoMapper configurations.

Fixed a typo in MinistryAdminBaseUpdateDto.

Rewrote unit tests for RegionAdmin to align with the new update flow.



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Streamlined admin update operations with refined data models for improved reliability and simplified data processing.
  - Enhanced security across admin endpoints by explicitly enforcing a specific authentication scheme for safer operations.

- **Chores**
  - Upgraded core authentication dependencies to the latest maintenance release to boost stability and performance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

synapsetron committed:
fix(AuthScheme): added berear auth scheme as a default schema for all controllers in AuthCommon

synapsetron committed:
refactor(ministry-admin): update logic to use MinistryAdminBaseUpdateDto for flexible updates

Replaced MinistryAdminBaseDto with MinistryAdminBaseUpdateDto in update methods to allow partial updates.
Refactored CommonMinistryAdminService to remove unused fields during updates.
Updated AutoMapper configuration and added a new DTO class to support the updated workflow.

synapsetron committed:
refactor(area-admin): use AreaAdminBaseUpdateDto for flexible updates

Replaced AreaAdminBaseDto with AreaAdminBaseUpdateDto in controller and service layers to support partial updates.
Adjusted IAreaAdminService and AreaAdminService to accept BaseUpdateUserDto as input.
Added new mapping profiles for AreaAdminBaseUpdateDto and created the DTO class.
Updated related unit tests and applied minor formatting and doc improvements.

synapsetron committed:
refactor(region-admin): update DTOs and mappings for partial updates

Replaced RegionAdminBaseDto with RegionAdminBaseUpdateDto in RegionAdminController and related tests to support flexible updates. Refactored IRegionAdminService and AreaAdminService to use BaseUpdateUserDto where appropriate. Introduced RegionAdminBaseUpdateDto to override inherited properties and disable [Required] validation. Updated AutoMapper profiles accordingly. Fixed typo in MinistryAdminBaseUpdateDto. Adjusted unit tests for updated DTOs and improved error handling.

DmyMi commented on OutOfSchool/Directory.Packages.props:
```diff
@@ -61,7 +61,7 @@
     <PackageVersion Include="OpenIddict.EntityFrameworkCore" Version="$(OpenIdDictVersion)" />
     <PackageVersion Include="OpenIddict.Quartz" Version="$(OpenIdDictVersion)" />
-    <PackageVersion Include="OpenIddict.Validation.AspNetCore" Version="$(OpenIdDictVersion)" />
+    <PackageVersion Include="OpenIddict.Validation.AspNetCore" Version="6.1.1" />
```
it is better to have unified version. So return the `"$(OpenIdDictVersion)"` variable usage here and change the variable value at the beginning of the file.

DmyMi commented on OutOfSchool/OutOfSchool.AuthServer.Tests/Controllers/RegionAdminControllerTests .cs:
```diff
@@ -172,14 +172,24 @@ public async Task Create_WithValidModel_ReturnsSuccessResponseDto()
     public async Task Update_WithValidModel_ReturnsSuccessResponseDto()
     {
         // Arrange
+        var updateDto = new RegionAdminBaseUpdateDto
```
The file name has extra space :) the contents of the file is fine.

DmyMi commented on OutOfSchool/OutOfSchool.BusinessLogic/Util/MappingProfile.cs:
```diff
@@ -489,6 +517,20 @@ public MappingProfile()
             .ForMember(dest => dest.EmailConfirmed, opt => opt.MapFrom(src => src.User.EmailConfirmed))
             .ForMember(dest => dest.UserName, opt => opt.MapFrom(src => src.User.UserName));
 
+        CreateMap<Parent, BaseUpdateUserDto>()
```
The `Update` DTO is going into the system. It should not be returned anywhere. So this mapping should not exist. So change the logic using this mapping to use `ShortUserDto` instead

synapsetron committed:
fix ( mapping ) : delete unnesesary maping profile to avoid returning of Update DTO
fix ( test) : remove extra space in test file name
fix ( package version): changed hardcoded version of openIdDict to variable
